rclpy/src/rclpy/_rclpy.c

```diff
@@ -370,17 +703,46 @@ rclpy_get_ready_subscriptions(PyObject * Py_UNUSED(self), PyObject * args)
     return NULL;
   }
 
-  size_t sub_idx;
-  PyObject * sub_ready_list = PyList_New(0);
-  for (sub_idx = 0; sub_idx < wait_set->size_of_subscriptions; sub_idx++) {
-    if (wait_set->subscriptions[sub_idx]) {
-      PyList_Append(
-        sub_ready_list,
-        PyLong_FromUnsignedLongLong((uint64_t)&wait_set->subscriptions[sub_idx]->impl));
+  size_t idx;
+  size_t idx_max;
```

Comment:
This PR will be merged as is, this function will be refactored later on to allow the repo transition.